doc/CMakeLists.txt

```diff
@@ -109,24 +114,86 @@ add_custom_target(
 )
 
 set(KI_SCRIPT ${ZEPHYR_BASE}/scripts/filter-known-issues.py)
+set(FIX_TEX_SCRIPT ${ZEPHYR_BASE}/doc/scripts/fix_tex.py)
 set(CONFIG_DIR ${ZEPHYR_BASE}/.known-issues/doc)
 
+#
+# HTML section
+#
 add_custom_target(
   html
   COMMAND ${CMAKE_COMMAND} -E env
   ZEPHYR_BUILD=${CMAKE_CURRENT_BINARY_DIR}
-  ${SPHINXBUILD} -w ${SPHINX_LOG} -N -t ${DOC_TAG} -b html ${ALLSPHINXOPTS} ${RST_OUT}/doc ${SPHINX_OUTPUT_DIR}
+  ${SPHINXBUILD} -w ${SPHINX_LOG} -N -t ${DOC_TAG} -b html ${ALLSPHINXOPTS} ${RST_OUT}/doc ${SPHINX_OUTPUT_DIR_HTML}
+  # Merge the Doxygen and Sphinx logs into a single file
+  COMMAND ${CMAKE_COMMAND} -P ${ZEPHYR_BASE}/cmake/util/fmerge.cmake ${DOC_LOG} ${DOXY_LOG} ${SPHINX_LOG}
+  COMMAND ${PYTHON_EXECUTABLE} ${KI_SCRIPT} --config-dir ${CONFIG_DIR} --errors ${DOC_WARN} --warnings ${DOC_WARN} ${DOC_LOG}
+  WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+  ${SPHINX_USES_TERMINAL}
+)
+
+#
+# LaTEX section
+#
+add_custom_target(
+  latex
+  COMMAND ${CMAKE_COMMAND} -E env
+  ZEPHYR_BUILD=${CMAKE_CURRENT_BINARY_DIR}
+  ${SPHINXBUILD} -w ${SPHINX_LOG} -N -t ${DOC_TAG} -b latex ${ALLSPHINXOPTS} ${RST_OUT}/doc ${SPHINX_OUTPUT_DIR_LATEX}
   # Merge the Doxygen and Sphinx logs into a single file
   COMMAND ${CMAKE_COMMAND} -P ${ZEPHYR_BASE}/cmake/util/fmerge.cmake ${DOC_LOG} ${DOXY_LOG} ${SPHINX_LOG}
   COMMAND ${PYTHON_EXECUTABLE} ${KI_SCRIPT} --config-dir ${CONFIG_DIR} --errors ${DOC_WARN} --warnings ${DOC_WARN} ${DOC_LOG}
+  COMMAND ${PYTHON_EXECUTABLE} ${FIX_TEX_SCRIPT} ${SPHINX_OUTPUT_DIR_LATEX}/zephyr.tex
   WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
   ${SPHINX_USES_TERMINAL}
 )
 
+#
+# PDF section
+#
+add_custom_command(
+  OUTPUT ${SPHINX_OUTPUT_DIR_LATEX}/zephyr.pdf
+  DEPENDS latexdocs ${SPHINX_OUTPUT_DIR_LATEX}/zephyr.tex
+  COMMAND make LATEXOPTS="-halt-on-error" LATEXMKOPTS="-quiet"
+  WORKING_DIRECTORY ${SPHINX_OUTPUT_DIR_LATEX}
+)
+
+if(NOT DEFINED SPHINX_OUTPUT_DIR)
+# Although latexmk allows specifying output directory,
+# makeindex fails if one is specified.
+# Hence the need of this to copy the PDF file over.
+add_custom_command(
+  OUTPUT ${SPHINX_OUTPUT_DIR_PDF}/zephyr.pdf
+  COMMAND mkdir -p -v ${SPHINX_OUTPUT_DIR_PDF}
```

Comment:
Please use cmake's built-in functionality to create a folder and copy files to it:
https://cmake.org/cmake/help/v3.8/manual/cmake.1.html#command-line-tool-mode
`cmake -E ...`

This makes  sure this will work on all operating systems.